service/worker/archiver/activities.go

```diff
@@ -123,9 +126,15 @@ func deleteHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 	if err == nil {
 		return nil
 	}
-	logger := tagLoggerWithHistoryRequest(tagLoggerWithActivityInfo(container.Logger, activity.GetInfo(ctx)), &request)
-	logger.Error("failed to delete history events", tag.Error(err))
-	if !common.IsPersistenceTransientError(err) {
+
+	if _, ok := err.(*serviceerror.WorkflowNotReady); !ok {
+		logger := tagLoggerWithHistoryRequest(tagLoggerWithActivityInfo(container.Logger, activity.GetInfo(ctx)), &request)
+		logger.Error("failed to delete workflow execution", tag.Error(err))
+	}
+
+	if !common.IsServiceTransientError(err) &&
+		!common.IsContextDeadlineExceededErr(err) &&
+		!common.IsContextCanceledErr(err) {
```

Comment:
WorkflowNotReady will pass this check and return errDeleteNonRetryable?
Also IsServiceTransientError() already checked IsContextDeadlineExceededErr and IsContextCanceledErr.